src/core/linalg/src/dense/4C_linalg_utlis_quaternion_interpolation.cpp

```diff
@@ -0,0 +1,564 @@
+// This file is part of 4C multiphysics licensed under the
+// GNU Lesser General Public License v3.0 or later.
+//
+// See the LICENSE.md file in the top-level for license information.
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include "4C_config.hpp"
+
+#include "4C_linalg_utlis_quaternion_interpolation.hpp"
+
+#include "4C_linalg_fixedsizematrix.hpp"
+#include "4C_linalg_utils_scalar_interpolation.hpp"
+#include "4C_utils_exceptions.hpp"
+
+#include <algorithm>
+#include <cassert>
+
+FOUR_C_NAMESPACE_OPEN
+
+/*--------------------------------------------------------------------*
+ *--------------------------------------------------------------------*/
+Core::LinAlg::Matrix<4, 1> Core::LinAlg::spherical_linear_interpolation(
+    const Core::LinAlg::Matrix<4, 1>& q1, const Core::LinAlg::Matrix<4, 1>& q2, double t)
+{
+  // Compute the dot product
+  double dot = q1.dot(q2);
+
+  // If the dot product is negative, slerp won't take the shorter path.
+  // Fix by reversing one quaternion.
+  Core::LinAlg::Matrix<4, 1> q2_mod = q2;
+  if (dot < 0.0)
+  {
+    for (int i = 0; i < 4; ++i) q2_mod(i, 0) = -q2_mod(i, 0);
+    dot = -dot;
+  }
+
+  const double linear_interpolation_threshold = 0.9995;
+  if (dot > linear_interpolation_threshold)
+  {
+    // If the quaternions are close, use linear interpolation
+    Core::LinAlg::Matrix<4, 1> result;
+    for (int i = 0; i < 4; ++i) result(i, 0) = (1.0 - t) * q1(i, 0) + t * q2_mod(i, 0);
```

Comment:
You could make the interpolation here obvious:
```suggestion
    for (int i = 0; i < 4; ++i) result(i, 0) = std::lerp(q1(i, 0), q2(i, 0), t);
```

Reply:
yes..

https://github.com/4C-multiphysics/4C/pull/876#issuecomment-2985630367